classlib/src/main/java/org/teavm/classlib/java/lang/TString.java

```diff
@@ -80,30 +81,34 @@ static TString fromArray(char[] characters) {
     private native void takeCharArray(char[] characters);
 
     public TString(byte[] bytes, int offset, int length, TString charsetName) throws TUnsupportedEncodingException {
-        this(bytes, offset, length, TCharset.forName(charsetName.toString()));
+        checkSIOOB(offset, length, bytes.length);
```

Comment:
I want also point out that there's hardly any code that depend on such checks. But checks like these have impact on performance and generated code size

Reply:
You can't never be sure. In my practice I have seen lots of code in different libraries which instead of doing preliminary checks, just does something like
```java
try {
// do stuff
} catch (IndexOutOfBoundsException e) {
// throw own wrapped exception
}
```
I understand that by default in teavm this checks are relaxed, but somebody can compile in strict mode when he relies on this libraries.

Reply:
It's still IOOBE.

Reply:
No, JDK throws StringIndexOutOfBoundsException. I'm not insisting on this change, but see comment above (with `catch StringIndexOutOfBoundsException`.

Reply:
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/String.html#%3Cinit%3E(char%5B%5D,int,int)

No mentions of this SIOOBE except for indexOf